File: amethyst/Events/Intermission.cs
using amethyst.Services;

namespace amethyst.Events;

public class IntermissionStarted(Guid7 id, IntermissionStartedBody body) 
    : Event<IntermissionStartedBody>(id, body)
    , IPeriodClockAligned;
public record IntermissionStartedBody(int DurationInSeconds);

public class IntermissionEnded(Guid7 id) : Event(id);

File: jamster.ui/packages/shared/src/types/events/Intermission.ts
import { EventWithoutBody } from "./Event";

export class IntermissionStarted extends EventWithoutBody {
    constructor() {
        super("IntermissionStarted");
    }
}

export class IntermissionEnded extends EventWithoutBody {
    constructor() {
        super("IntermissionEnded");
    }
}